components/courses/app/models/courses/lecture.rb

```diff
@@ -17,5 +17,11 @@ class Lecture < ApplicationRecord
     enum status: [UPCOMING, PASSED]
 
     validates :title, :started_at, :finished_at, :venue_id, :mentor_id, presence: true
+    validate  :time_policy
```

Comment:
Also, this policy can be improved in future to check if time range is okay. For overlapping lectures, too long lectures, etc.